src/main/scala/amba/ahb/RegisterRouter.scala

```diff
@@ -64,15 +76,49 @@ case class AHBRegisterNode(address: AddressSet, concurrency: Int = 0, beatBytes:
 
     out.ready := Bool(true)
     assert (d_phase || !out.valid)
+
+    genRegDescsJson(mapping:_*)
+    genOMRegMap(mapping:_*)
+  }
+    def genOMRegMap(mapping: RegField.Map*): OMRegisterMap = {
+    OMRegister.convert(mapping = mapping:_*)
+  }
+
+  def genRegDescsJson(mapping: RegField.Map*) {
+    // Dump out the register map for documentation purposes.
+    val base = address.base
+    val baseHex = s"0x${base.toInt.toHexString}"
+    val name = s"deviceAt${baseHex}" //TODO: It would be better to name this other than "Device at ...."
+    val json = GenRegDescsAnno.serialize(base, name, mapping:_*)
+    var suffix = 0
+    while( ElaborationArtefacts.contains(s"${baseHex}.${suffix}.regmap.json")) {
+      suffix = suffix + 1
+    }
+    ElaborationArtefacts.add(s"${baseHex}.${suffix}.regmap.json", json)
+
+    val module = Module.currentModule.get.asInstanceOf[RawModule]
+    GenRegDescsAnno.anno(
+      module,
+      base,
+      mapping:_*)
   }
 }
 
 // These convenience methods below combine to make it possible to create a AHB
 // register mapped device from a totally abstract register mapped device.
 
-abstract class AHBRegisterRouterBase(address: AddressSet, interrupts: Int, concurrency: Int, beatBytes: Int, undefZero: Boolean, executable: Boolean)(implicit p: Parameters) extends LazyModule
+abstract class AHBRegisterRouterBase(devname: String, devcompat: Seq[String], address: AddressSet, interrupts: Int, concurrency: Int, beatBytes: Int, undefZero: Boolean, executable: Boolean)(implicit p: Parameters) extends LazyModule
```

Comment:
This API for the register router is supposed to be deprecated. That said, I understand that some existing code may still be using it.